src/datetime.rs

```diff
@@ -131,6 +131,28 @@ impl crate::DateTime {
         Self::from_millis(dt.timestamp_millis())
     }
 
+    /// Convert the given year, month, date, hour, minute, second and millisecond into a
+    /// `bson::DateTime`, truncating it to millisecond precision.
+    pub fn from_specific_datetime(
+        year: i32,
+        month: u8,
+        day: u8,
+        hour: u8,
+        min: u8,
+        sec: u8,
+        millis: u16,
+    ) -> Result<Self> {
```

Comment:
I think the intent of this ticket was to have a separate method for each time increment listed here but is that what you were thinking @patrickfreed? Sorry if this was unclear!